### PR TITLE
changed heading of table to required from optional

### PR DESCRIPTION
**Pull request by WAvdBeek**

### CODE REVIEW

File: sdf.md [hunk 1]
```diff
@@ -137,15 +136,14 @@ The odmObject keyword denotes zero or more Object definitions. A object may cont
...
-|id| integer | yes | internal unique identifier for the definition |
-|name|string|yes|human readable name|
-|description|string|yes|human readable description|
-|title|string|yes|human readable title to display|
-|optional| boolean|yes|defines whether this element is optional in an implementation|
-|include|array|yes|reference to definitions to be included|
-| odmType|object|yes|reference to a definition to be used as a template for a new definition|
+| Quality | Type | Required | Description | Default |
+|---|---|---|---|---|
+|id| integer | yes | internal unique identifier for the definition |  N/A |
```
Comment (akeranen) [inline sdf.md]: This makes ID required. Which I think makes sense, but is not a simple polarity change as for the others.

File: sdf.md [hunk 2]
```diff
@@ -137,15 +136,14 @@ The odmObject keyword denotes zero or more Object definitions. A object may cont
...
+|name|string|no|human readable name| N/A |
+|description|string|no|human readable description| N/A |
```
Comment (akeranen) [inline sdf.md]: Probably description (of some sort) should be required, but maybe that's better handled and discussed in a separate PR.

File: sdf.md [hunk 3]
```diff
@@ -165,34 +163,33 @@ Properties are used to model elements of state.
...
-|---|---|---|---|
-|id| integer | yes | internal unique identifier for the definition |
-|name|string|yes|human readable name|
-|description|string|yes|human readable description|
-|title|string|yes|human readable title to display|
-|optional| boolean|yes|defines whether this element is optional in an implementation|
+| Quality | Type | Required | Description | Default |
+|---|---|---|---|---|
+|id| integer | yes | internal unique identifier for the definition | N/A |
+|name|string|yes|human readable name| N/A |
```
Comment (akeranen) [inline sdf.md]: Now name, desc, title are required for property but not for object. Maybe we want to keep these consistent for now and then discuss in a separate issue what should be required and what not to have all that documented in a single place.

File: sdf.md [hunk 4]
```diff
@@ -165,34 +163,33 @@ Properties are used to model elements of state.
...
-|minimum|number|yes|lower limit of value in the representation format|
-|maximum|number|yes|upper limit of value in the representation format|
-|multipleOf|number|yes|indicates the resolution of the number in representation format|
-|enum|array|yes|enumeration constraint|
-|pattern|string|yes|regular expression to constrain a string pattern|
-|minLength|integer|yes|shortest length string in octets|
-|maxLength|integer|yes|longest length string in octets|
-|default|number, boolean, string|yes|specifies the default value for initialization|
-|const|number, boolean, string|yes|specifies a constant value for a data item or property|
+|type|object|yes|reference to a definition to be used as a template for a new definition| N/A |
```
Comment (akeranen) [inline sdf.md]: type, readOnly, writeOnly, observable, etc. should probably *not* be required

File: sdf.md [hunk 5]
```diff
@@ -209,15 +206,15 @@ Actions are used to model commands and methods which are invoked. Actions have p
...
+| Quality | Type | Required | Description |
 |---|---|---|---|
-|id| integer | yes | internal unique identifier for the definition |
-|name|string|yes|human readable name|
-|description|string|yes|human readable description|
-|title|string|yes|human readable title to display|
-|optional| boolean|yes|defines whether this element is optional in an implementation|
-|include|array|yes|reference to definitions to be included|
-|type|object|yes|reference to a definition to be used as a template for a new definition|
+|id| integer | no | internal unique identifier for the definition |
```
Comment (akeranen) [inline sdf.md]: For consistency, Required here should probably be the same now as for others.

File: sdf.md [hunk 6]
```diff
@@ -234,9 +231,9 @@ Events are used to model asynchronous occurrences that may be communicated proac
 
 - Qualities of odmEvent
 
-| Quality | Type | Optional | Description |
+| Quality | Type | Required | Description |
 |---|---|---|---|
-|id| integer | yes | internal unique identifier for the definition |
+|id| integer | no | internal unique identifier for the definition |
 |name|string|yes|human readable name|
```
Comment (akeranen) [inline sdf.md]: Again, good to check for consistency for required here.

File: sdf.md [hunk 7]
```diff
@@ -261,30 +258,30 @@ odmData is used for Action parameters, for Event data, and for reusable constrai
...
+|title|string|no|human readable title to display|
+|optional| boolean|no|defines whether this element is optional in an implementation|
+|include|array|no|reference to definitions to be included|
+|type|object|no|reference to a definition to be used as a template for a new definition|
+|units|string|no|[SenML unit][] code|
+|nullable|boolean|no|indicates a null value is available for this type|
+|encoding|map|no|applies additional constraints|
+|scaleMinimum|number|no|lower limit of value in units|
+|scaleMaximum|number|no|upper limit of value in units|
 |type|string, enum|yes|JSON data type|
```
Comment (akeranen) [inline sdf.md]: intentionally "yes" for required here?

File: sdf.md (hunk 4 above)
Comment (WAvdBeek) [reply sdf.md]: now not required anymore

File: sdf.md (hunk 5 above)
Comment (WAvdBeek) [reply sdf.md]: fixed consistency: all id's are now required

**Pull request merged**